### ISSUE TITLE
Invalid XML declarations are slippign through the cracks. 

### ISSUE THREAD

**Issue by PokerPro123** [IDPF/epubcheck#842]
Tested in opf and html files:

Passes validation:
<?xxml version="1.0" encoding="utf-8" standalone="yes"?>
<?xl version="1.0" encoding="utf-8" standalone="no"?>

**Comment by AntonBolfing** [comment]
Besten Dank für Ihre Nachricht.

Ich bin bis am Dienstag, dem 15. Mai 2018 abwesend und kann Ihre Nachricht erst ab dann wieder bearbeiten. Ihre Email wird nicht weitergeleitet.


Beste Grüsse

Anton Bolfing

*******************************************************

Thank you for your message.

I'm out of office until Tuesday, May 15, 2018.  I will not respond to your email until then.

Best regards

Anton Bolfing

**Comment by mattgarrish** [comment]
There's actually nothing technically wrong here. You can have processing instructions to start a file, and when the xml declaration is misnamed it just becomes a regular old processing instruction.

The XML declaration doesn't really do anything in EPUB, either, since version 1.0 and UTF-8 are the defaults, so the lack of a correct xml declaration isn't going to negatively affect the rendering.

I haven't heard of anyone releasing EPUBs with processing instructions still in their documents, though, so it probably would be harmless to warn if any are found in the document prolog (ignoring actual xml declarations, of course).

**Comment by PokerPro123** [comment]
So XML parsers are supposed to ignore this?

**Comment by murata2makoto** [comment]
Such a PI is certainly correct as a part of XML.  In my understanding, neither WHATWG HTML nor W3C HTML 5.2 disallows the use of such PIs.  So, such PIs are perfectly legitimate, as I see it.  WHATWG DOM allows PIs.  I thus conclude that your example (<?xl ...>) will appear in the DOM tree as a PI, but is not used for rendering.

**Comment by mattgarrish** [comment]
> neither WHATWG HTML nor W3C HTML 5.2 disallows the use of such PIs

No, but they're also generally useless. XML style sheet PIs haven't gone anywhere in epub, so other than relics from internal processes and applications what function would they serve? In all cases, there's no harm in telling someone they're an odd choice in the output.

But I would keep the warning at the info level if implemented, as they are legitimate.

**Comment by rkwright** [comment]
So isn't this an enhancement and not a bug?

**Comment by tofi86** [comment]
> So isn't this an enhancement and not a bug?

I agree.

> But I would keep the warning at the info level if implemented, as they are legitimate.

But there currently is no warning. So what to do? Create one? I don't think it's really necessary...

**Comment by rdeltour** [comment]
Yeah, I would be tempted to just do nothing on that one: EpubCheck is a validation tool, not a best practice tool.

That said, if several people upvote this issue, I guess it doesn't harm to implement that as an INFO-level. But if the usage is rare, I doubt it's worth implementing it?
